### PR TITLE
Collect Sector

### PR DESCRIPTION
**Pull request by pedromcp90**
Added the sector selector 
https://user-images.githubusercontent.com/1659534/189894033-10d59b16-9e69-470a-9c13-84e948c02d1f.mp4

### CODE REVIEW

File: config.php [hunk 1]
```diff
@@ -43,6 +43,16 @@
...
+
+$action = Tools::getValue('action');
+if($action === 'set_sector'){
+    $sector_data = Tools::getValue('sector_shop');
+    $module->saveSectorData($sector_data);
+    echo json_encode(["success" => true]);
+    exit;
+}
+
+
```
Comment (ivan-doofinder) [inline config.php]: You can't receive a form here, it's insecure.
You have to do it the same as the rest of the forms in the module.

**Review by ivan-doofinder — CHANGES_REQUESTED** [review]

File: config.php (hunk 1 above)
Comment (pedromcp90) [reply config.php]: OK, I placed this here because we were also receiving the API connection details here, should we move also the API connection form?

Comment (ivan-doofinder) [reply config.php]: Currently the API KEY is received from Doofinder in an endpoint created for it with a temporary security token (config.php file).

This case is an internal form of the module. Prestashop already offers us the option to manage the forms, as it is done in the rest of the forms of the module.

File: doofinder.php [hunk 2]
```diff
@@ -338,6 +367,25 @@ protected function isConfigured()
         return Db::getInstance()->getValue($sql);
     }
 
+    protected function isSectorConfigured($shop_id)
+    {
+        $shops = Shop::getShops();
+        usort($shops, function ($a, $b) {
+            return $a['id_shop'] - $b['id_shop'];
+        });
```
Comment (abraham-doofinder) [inline doofinder.php]: You can use `getSortedShops()` in this case

Comment (pedromcp90) [reply doofinder.php]: Thanks, I missed replacing that one.

**Review by abraham-doofinder — APPROVED** [review]

**Pull request closed without merging**